src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbol.cs

```diff
@@ -1558,11 +1558,23 @@ internal override void AddSynthesizedAttributes(ModuleCompilationState compilati
         /// Checks to see if a body is legal given the current modifiers.
         /// If it is not, a diagnostic is added with the current type.
         /// </summary>
-        protected void CheckModifiersForBody(Location location, DiagnosticBag diagnostics)
+        protected void CheckModifiersForBody(SyntaxNode syntax, Location location, DiagnosticBag diagnostics)
         {
             if (_containingType.IsInterface)
             {
-                diagnostics.Add(ErrorCode.ERR_InterfaceMemberHasBody, location, this);
+                if (!this.IsAccessor())
+                {
+                    Binder.CheckFeatureAvailability(syntax, MessageID.IDS_DefaultInterfaceImplementation, diagnostics, location);
+
+                    if (!ContainingAssembly.RuntimeSupportsDefaultInterfaceImplementation)
+                    {
+                        diagnostics.Add(ErrorCode.ERR_RuntimeDoesNotSupportDefaultInterfaceImplementation, location);
+                    }
+                }
+                else
+                {
+                    diagnostics.Add(ErrorCode.ERR_InterfaceMemberHasBody, location, this);
+                }
```

Comment:
It might be helpful to add a quick note of what is supported and what is NYI. A small example might be enough. I was also under impression that the change enables more than literally stated in the subject.